opentelemetry-exporter-gcp-logging/src/opentelemetry/exporter/cloud_logging/__init__.py

```diff
@@ -0,0 +1,226 @@
+# from
+# https://github.com/googleapis/python-logging/blob/5309478c054d0f2b9301817fd835f2098f51dc3a/google/cloud/logging_v2/services/logging_service_v2/transports/grpc.py#L179-L182
+_OPTIONS = [
+    ("grpc.max_send_message_length", -1),
+    ("grpc.max_receive_message_length", -1),
+    ("grpc.primary_user_agent", _USER_AGENT),
+]
+
+# severityMapping maps the integer severity level values from OTel [0-24]
+# to matching Cloud Logging severity levels.
+SEVERITY_MAPPING: dict[int, int] = {
+    0: LogSeverity.DEFAULT,  # Default, 0
+    1: LogSeverity.DEBUG,  #
+    2: LogSeverity.DEBUG,  #
+    3: LogSeverity.DEBUG,  #
+    4: LogSeverity.DEBUG,  #
+    5: LogSeverity.DEBUG,  #
+    6: LogSeverity.DEBUG,  #
+    7: LogSeverity.DEBUG,  #
+    8: LogSeverity.DEBUG,  # 1-8 -> Debug
+    9: LogSeverity.INFO,  #
+    10: LogSeverity.INFO,  # 9-10 -> Info
+    11: LogSeverity.NOTICE,  #
+    12: LogSeverity.NOTICE,  # 11-12 -> Notice
+    13: LogSeverity.WARNING,  #
+    14: LogSeverity.WARNING,  #
+    15: LogSeverity.WARNING,  #
+    16: LogSeverity.WARNING,  # 13-16 -> Warning
+    17: LogSeverity.ERROR,  #
+    18: LogSeverity.ERROR,  #
+    19: LogSeverity.ERROR,  #
+    20: LogSeverity.ERROR,  # 17-20 -> Error
+    21: LogSeverity.CRITICAL,  #
+    22: LogSeverity.CRITICAL,  # 21-22 -> Critical
+    23: LogSeverity.ALERT,  # 23 -> Alert
+    24: LogSeverity.EMERGENCY,  # 24 -> Emergency
+}
+
+
+class CloudLoggingExporter(LogExporter):
+    def __init__(
+        self,
+        project_id: Optional[str] = None,
+        default_log_name: Optional[str] = None,
+        client: Optional[LoggingServiceV2Client] = None,
+    ):
+        self.project_id: str
+        if not project_id:
+            _, default_project_id = google.auth.default()
+            self.project_id = str(default_project_id)
+        else:
+            self.project_id = project_id
+        if default_log_name:
+            self.default_log_name = default_log_name
+        else:
+            self.default_log_name = "otel_python_inprocess_log_name_temp"
+        if default_log_name:
+            self.default_log_name = default_log_name
+        else:
+            self.default_log_name = "otel_python_inprocess_log_name_temp"
+        self.client = client or LoggingServiceV2Client(
+            transport=LoggingServiceV2GrpcTransport(
+                channel=LoggingServiceV2GrpcTransport.create_channel(
+                    options=_OPTIONS,
+                )
+            )
+        )
+        self.service_resource_labels = True
+
+    def export(self, batch: Sequence[LogData]):
+        log_entries = []
+        for log_data in batch:
+            log_record = log_data.log_record
+            attributes = log_record.attributes or {}
+            project_id = self.project_id
+            if attributes.get(PROJECT_ID_ATTRIBUTE_KEY):
+                project_id = str(attributes.get(PROJECT_ID_ATTRIBUTE_KEY))
```

Comment:
nit I think this pattern would be a little more readable throughout here
```suggestion
            project_id = attributes.get(PROJECT_ID_ATTRIBUTE_KEY, self.project_id)
```